### ISSUE TITLE
model validationAttributeNames

### ISSUE THREAD

**Issue by acodercat** [dwightwatson/validating#199]
I wanna in my model like this
```php 
public $validationAttributeNames = [
        'starts_on' => 'Starts On Date',
        'ends_on' => 'Ends On Date',
    ];
```

**Comment by dwightwatson** [comment]
I don’t understand your question?

**Comment by acodercat** [comment]
in model I wanna define validationAttribute

**Comment by dwightwatson** [comment]
You can... https://github.com/dwightwatson/validating/blob/master/src/ValidatingTrait.php#L35-L40

**Comment by acodercat** [comment]
"App\Family\Domain\Model\FamilyMemberModel and Watson\Validating\ValidatingTrait define the same property ($validationAttributeNames) in the composition of App\Family\Domain\Model\FamilyMemberModel. However, the definition differs and is considered incompatible. Class was composed"

**Comment by dwightwatson** [comment]
Could be because you're using `public` instead of `protected`. You're really going to have to supply more than a fraction of a code sample of you want help

**Comment by acodercat** [comment]
```php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: codercat
 * Date: 2018/5/21
 * Time: 17:31
 */

namespace App\Family\Domain\Model;

use Watson\Validating\ValidatingTrait;
use App\Common\Model\BaseModel;
use App\Family\Domain\Exceptions\DuplicateBindException;

class FamilyMemberModel extends BaseModel
{

    use ValidatingTrait;


    protected $table = 'family_member';
    protected $fillable = ['owner_id', 'age', 'owner_relationship', 'sex', 'mobile', 'fullname','id_card_no','health_card_no','is_auth'];

    protected $rules = [
        'mobile'   => 'bail|required|mobile',
        'fullname'   => 'bail|required',
        'sex'   => 'bail|required|in:male,female',
        'age'   => 'bail|nullable|integer|between:0,130',
        'owner_relationship'   => 'bail|nullable|in:self,children,other,husband,wife,mother,daughter,father',
        'id_card_no'   => 'bail|nullable|id_card_no|unique:family_member',

    ];

    protected $validationAttributeNames = [
        'mobile' => '手机号',
        'fullname' => '姓名',
    ];
}
```

**Comment by acodercat** [comment]
thanks

**Comment by dwightwatson** [comment]
Did switching to `protected` solve the issue? If not could you provide the complete stack trace of the error.

**Comment by acodercat** [comment]
How to do switching to protected

**Comment by acodercat** [comment]
$validationAttributeNames is protected

**Comment by dwightwatson** [comment]
In your initial example you had it as `public`, when it was supposed to be `protected`. Now that you've changed it to `protected` does the problem you were having still persist?

**Comment by acodercat** [comment]
error :"App\Family\Domain\Model\FamilyMemberModel and Watson\Validating\ValidatingTrait define the same property ($validationAttributeNames) in the composition of App\Family\Domain\Model\FamilyMemberModel. However, the definition differs and is considered incompatible. Class was composed"

**Comment by dwightwatson** [comment]
Cool, I understand what you're running into now. Found the issue and pushed a fix in `3.1.4`. It might take a little while to become available on Composer but when you pull in the new version you should be good to go. Let me know if it doesn't resolve the issue for you.
